Revamp Service Bindings docs for RPC (#13252)
cloudflare/cloudflare-docs: content/workers/_partials/_service-binding-rpc-example.md

---
_build:
  publishResources: false
  render: never
  list: never
---

For example, if Worker B implements the public method `add(a, b)`:

```toml
---
filename: wrangler.toml
---
name = "worker_b"
main = "./src/workerB.js"
```

```js
---

> filename: workerB.js

Reviewer: ```suggestion
filename: src/workerB.js
```

---
import { WorkerEntrypoint } from "cloudflare:workers";

> export class WorkerB extends WorkerEntrypoint {
>   async add(a, b) { return a + b; }
> }

Reviewer: I think it's more interesting to show it without the async. Also, you still need to export a handler (e.g. `fetch`) on the default export of the service.

```suggestion
export default class extends WorkerEntrypoint {
  async fetch() { return new Response("Hello from Worker B"); }
  
  add(a, b) { return a + b; } 
}
```

```

Worker A can declare a [binding](/workers/runtime-apis/bindings) to Worker B:

```toml
---
filename: wrangler.toml
---
name = "worker_a"
main = "./src/workerA.js"
services = [
  { binding = "WORKER_B", service = "worker_b" }
]
```

Making it possible for Worker A to call the `add()` method from Worker B:

```js
---

> filename: workerA.js

Reviewer: ```suggestion
filename: src/workerA.js
```
